## Final Project Submission

Please fill out:
* Student name: 
* Student pace: self paced / part time / full time
* Scheduled project review date/time: 
* Instructor name: 
* Blog post URL:


In [ ]:
# Your code here - remember to use markdown cells for comments as well!

In [2]:
import pandas as pd

In [2]:
df = pd.read_csv('rt.movie_info_info.tsv', sep='\t', error_bad_lines=False)
df.head()

id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                            writer  theater_date      dvd_date currency  \
0                   Ernest Tidyman   Oct 9, 1971  Sep 25, 2001      NaN   
1     David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
2                   Allison Anders  Sep 13, 1996  Apr 18, 2000      NaN   
3  Paul Attanasio|Michael Crichton   Dec 9, 1994  Aug 27, 1997      NaN   
4                     Giles Cooper           NaN           NaN      NaN   

  box_office      runtime             studio  
0        NaN  104 minutes                NaN  
1    600,000  108 minutes  Entertainment One  
2        NaN  116 minutes                NaN  
3        NaN  128 minutes                NaN  
4        NaN  200 minutes                NaN

In [3]:
df.tail()

id                                           synopsis rating  \
1555  1996  Forget terrorists or hijackers -- there's a ha...      R   
1556  1997  The popular Saturday Night Live sketch was exp...     PG   
1557  1998  Based on a novel by Richard Powell, when the l...      G   
1558  1999  The Sandlot is a coming-of-age story about a g...     PG   
1559  2000  Suspended from the force, Paris cop Hubert is ...      R   

                                                  genre            director  \
1555   Action and Adventure|Horror|Mystery and Suspense                 NaN   
1556                 Comedy|Science Fiction and Fantasy        Steve Barron   
1557  Classics|Comedy|Drama|Musical and Performing Arts      Gordon Douglas   
1558    Comedy|Drama|Kids and Family|Sports and Fitness  David Mickey Evans   
1559  Action and Adventure|Art House and Internation...                 NaN   

                                                writer  theater_date  \
1555                                               NaN  Aug 18, 2006   
1556  Terry Turner|Tom Davis|Dan Aykroyd|Bonnie Turner  Jul 23, 1993   
1557                                               NaN   Jan 1, 1962   
1558                  David Mickey Evans|Robert Gunter   Apr 1, 1993   
1559                                        Luc Besson  Sep 27, 2001   

          dvd_date currency  box_office      runtime             studio  
1555   Jan 2, 2007        $  33,886,034  106 minutes    New Line Cinema  
1556  Apr 17, 2001      NaN         NaN   88 minutes  Paramount Vantage  
1557  May 11, 2004      NaN         NaN  111 minutes                NaN  
1558  Jan 29, 2002      NaN         NaN  101 minutes                NaN  
1559  Feb 11, 2003      NaN         NaN   94 minutes  Columbia Pictures

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 12 columns):
id              1560 non-null int64
synopsis        1498 non-null object
rating          1557 non-null object
genre           1552 non-null object
director        1361 non-null object
writer          1111 non-null object
theater_date    1201 non-null object
dvd_date        1201 non-null object
currency        340 non-null object
box_office      340 non-null object
runtime         1530 non-null object
studio          494 non-null object
dtypes: int64(1), object(11)
memory usage: 146.4+ KB


In [5]:
df.isna().sum()

id                 0
synopsis          62
rating             3
genre              8
director         199
writer           449
theater_date     359
dvd_date         359
currency        1220
box_office      1220
runtime           30
studio          1066
dtype: int64

In [23]:
df.dropna(subset=['box_office'])

id                                           synopsis rating  \
1        3  New York City, not-too-distant-future: Eric Pa...      R   
6       10  Some cast and crew from NBC's highly acclaimed...  PG-13   
7       13  Stewart Kane, an Irishman living in the Austra...      R   
8       14  "Love Ranch" is a bittersweet love story that ...      R   
15      22  Two-time Academy Award Winner Kevin Spacey giv...      R   
...    ...                                                ...    ...   
1541  1980  A band of renegades on the run in outer space ...  PG-13   
1542  1981  Money, Fame and the Knowledge of English. In I...     NR   
1545  1985  A woman who joins the undead against her will ...      R   
1546  1986  Aki Kaurismaki's The Man Without a Past opens ...     PG   
1555  1996  Forget terrorists or hijackers -- there's a ha...      R   

                                                 genre             director  \
1                    Drama|Science Fiction and Fantasy     David Cronenberg   
6                                               Comedy          Jake Kasdan   
7                                                Drama         Ray Lawrence   
8                                                Drama      Taylor Hackford   
15                   Comedy|Drama|Mystery and Suspense  George Hickenlooper   
...                                                ...                  ...   
1541  Action and Adventure|Science Fiction and Fantasy          Joss Whedon   
1542                                      Comedy|Drama         Gauri Shinde   
1545                       Horror|Mystery and Suspense  Sebastian Gutierrez   
1546          Art House and International|Comedy|Drama                  NaN   
1555  Action and Adventure|Horror|Mystery and Suspense                  NaN   

                                writer  theater_date      dvd_date currency  \
1         David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
6                           Mike White  Jan 11, 2002  Jun 18, 2002        $   
7     Raymond Carver|Beatrix Christian  Apr 27, 2006   Oct 2, 2007        $   
8                        Mark Jacobson  Jun 30, 2010   Nov 9, 2010        $   
15                       Norman Snider  Dec 17, 2010   Apr 5, 2011        $   
...                                ...           ...           ...      ...   
1541                       Joss Whedon  Sep 30, 2005  Dec 20, 2005        $   
1542                      Gauri Shinde   Oct 5, 2012  Nov 20, 2012        $   
1545               Sebastian Gutierrez   Jun 1, 2007   Oct 9, 2007        $   
1546                               NaN  Aug 30, 2002   Oct 7, 2003        $   
1555                               NaN  Aug 18, 2006   Jan 2, 2007        $   

      box_office      runtime                  studio  
1        600,000  108 minutes       Entertainment One  
6     41,032,915   82 minutes      Paramount Pictures  
7        224,114  123 minutes  Sony Pictures Classics  
8        134,904  117 minutes                     NaN  
15     1,039,869  108 minutes            ATO Pictures  
...          ...          ...                     ...  
1541  25,335,935  119 minutes      Universal Pictures  
1542   1,416,189  129 minutes      Eros Entertainment  
1545      59,371   98 minutes        IDP Distribution  
1546     794,306   97 minutes                     NaN  
1555  33,886,034  106 minutes         New Line Cinema  

[340 rows x 12 columns]

In [6]:
df.columns

Index(['id', 'synopsis', 'rating', 'genre', 'director', 'writer',
       'theater_date', 'dvd_date', 'currency', 'box_office', 'runtime',
       'studio'],
      dtype='object')

In [7]:
df_rt_edit = df.dropna(subset=['box_office'])

In [8]:
df_rt_edit.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 340 entries, 1 to 1555
Data columns (total 12 columns):
id              340 non-null int64
synopsis        340 non-null object
rating          340 non-null object
genre           340 non-null object
director        299 non-null object
writer          273 non-null object
theater_date    334 non-null object
dvd_date        334 non-null object
currency        340 non-null object
box_office      340 non-null object
runtime         338 non-null object
studio          305 non-null object
dtypes: int64(1), object(11)
memory usage: 34.5+ KB


# rt.movie_info_info.tsv
 

In [16]:
df_reviews = pd.read_csv('rt.reviews.tsv', sep='\t', error_bad_lines=False, encoding='latin1')
df_reviews.head(15)

id                                             review rating   fresh  \
0    3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1    3  It's an allegory in search of a meaning that n...    NaN  rotten   
2    3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3    3  Continuing along a line introduced in last yea...    NaN   fresh   
4    3             ... a perverse twist on neorealism...     NaN   fresh   
5    3  ... Cronenberg's Cosmopolis expresses somethin...    NaN   fresh   
6    3  Quickly grows repetitive and tiresome, meander...      C  rotten   
7    3  Cronenberg is not a director to be daunted by ...    2/5  rotten   
8    3  Cronenberg's cold, exacting precision and emot...    NaN   fresh   
9    3  Over and above its topical urgency or the bit ...    NaN   fresh   
10   3  Cosmopolis" is a hypnotic examination of our m...    NaN   fresh   
11   3  While not one of Cronenberg's stronger films, ...     B-   fresh   
12   3  Robert Pattinson works mighty hard to make Cos...    2/4  rotten   
13   3  The anger over the injustice of the financial ...      B   fresh   
14   3  For those who like their Cronenberg thick and ...    3/4   fresh   

             critic  top_critic            publisher                date  
0        PJ Nabarro           0      Patrick Nabarro   November 10, 2018  
1    Annalee Newitz           0              io9.com        May 23, 2018  
2      Sean Axmaker           0     Stream on Demand     January 4, 2018  
3     Daniel Kasman           0                 MUBI   November 16, 2017  
4               NaN           0         Cinema Scope    October 12, 2017  
5   Michelle Orange           0     Capital New York  September 11, 2017  
6    Eric D. Snider           0      EricDSnider.com       July 17, 2013  
7      Matt Kelemen           0   Las Vegas CityLife      April 21, 2013  
8      Sean Axmaker           0        Parallax View      March 24, 2013  
9      Kong Rithdee           0         Bangkok Post       March 4, 2013  
10     Tim Grierson           0             Deadspin   February 12, 2013  
11     Emanuel Levy           0      EmanuelLevy.Com    February 3, 2013  
12   Christian Toto           0        Big Hollywood    January 15, 2013  
13     Robert Roten           0  Laramie Movie Scope     January 7, 2013  
14      Marty Mapes           0          Movie Habit    October 20, 2012

In [11]:
df_reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54432 entries, 0 to 54431
Data columns (total 8 columns):
id            54432 non-null int64
review        48869 non-null object
rating        40915 non-null object
fresh         54432 non-null object
critic        51710 non-null object
top_critic    54432 non-null int64
publisher     54123 non-null object
date          54432 non-null object
dtypes: int64(2), object(6)
memory usage: 3.3+ MB


In [12]:
df_reviews.columns

Index(['id', 'review', 'rating', 'fresh', 'critic', 'top_critic', 'publisher',
       'date'],
      dtype='object')

In [13]:
df_reviews.isna().sum()

id                0
review         5563
rating        13517
fresh             0
critic         2722
top_critic        0
publisher       309
date              0
dtype: int64

In [15]:
df_reviews.tail(15)

id                                             review rating   fresh  \
54417  2000  The funny thing is, I didn't mind all this con...    NaN   fresh   
54418  2000  Slight, but colorful and forward-moving... Was...     C+   fresh   
54419  2000            Sleek, shallow, but frequently amusing.  2.5/4   fresh   
54420  2000  The spaniel-eyed Jean Reno infuses Hubert with...    3/4   fresh   
54421  2000  Manages to be somewhat well-acted, not badly a...  1.5/4  rotten   
54422  2000  Arguably the best script that Besson has writt...  3.5/5   fresh   
54423  2000          Despite the title, Wasabi isn't so spicy.      C  rotten   
54424  2000  Dawdles and drags when it should pop; it doesn...  1.5/5  rotten   
54425  2000  Despite Besson's high-profile name being Wasab...    NaN   fresh   
54426  2000  The film lapses too often into sugary sentimen...    NaN  rotten   
54427  2000  The real charm of this trifle is the deadpan c...    NaN   fresh   
54428  2000                                                NaN    1/5  rotten   
54429  2000                                                NaN    2/5  rotten   
54430  2000                                                NaN  2.5/5  rotten   
54431  2000                                                NaN    3/5   fresh   

                  critic  top_critic               publisher  \
54417      Andrew Sarris           1                Observer   
54418        Brent Simon           0     Entertainment Today   
54419       Gene Seymour           1                 Newsday   
54420       Megan Turner           1           New York Post   
54421        Bob Strauss           0  Los Angeles Daily News   
54422         Wade Major           0      Boxoffice Magazine   
54423                NaN           0               E! Online   
54424     Manohla Dargis           1       Los Angeles Times   
54425         Andy Klein           0               New Times   
54426       Paul Malcolm           1             L.A. Weekly   
54427      Laura Sinagra           1           Village Voice   
54428  Michael Szymanski           0              Zap2it.com   
54429       Emanuel Levy           0         EmanuelLevy.Com   
54430   Christopher Null           0          Filmcritic.com   
54431    Nicolas Lacroix           0            Showbizz.net   

                     date  
54417     October 2, 2002  
54418  September 30, 2002  
54419  September 27, 2002  
54420  September 27, 2002  
54421  September 27, 2002  
54422  September 27, 2002  
54423  September 27, 2002  
54424  September 26, 2002  
54425  September 26, 2002  
54426  September 26, 2002  
54427  September 24, 2002  
54428  September 21, 2005  
54429       July 17, 2005  
54430   September 7, 2003  
54431   November 12, 2002

In [18]:
df_reviews_edit = df_reviews.dropna(subset=['rating'])

In [24]:
df_reviews_edit.head(15)

id                                             review rating   fresh  \
0    3  A distinctly gallows take on contemporary fina...    3/5   fresh   
6    3  Quickly grows repetitive and tiresome, meander...      C  rotten   
7    3  Cronenberg is not a director to be daunted by ...    2/5  rotten   
11   3  While not one of Cronenberg's stronger films, ...     B-   fresh   
12   3  Robert Pattinson works mighty hard to make Cos...    2/4  rotten   
13   3  The anger over the injustice of the financial ...      B   fresh   
14   3  For those who like their Cronenberg thick and ...    3/4   fresh   
15   3  For better or worse - often both - Cosmopolis ...    3/5   fresh   
16   3  For one of the smartest films I've seen in a w...    4/5   fresh   
17   3      Its major problem is that it's not cinematic.      B   fresh   
18   3  It's fascinating to watch Pattinson actually a...    2/4  rotten   
19   3  A black comedy as dry and deadpan as a bleache...    4/4   fresh   
20   3  While 'Cosmopolis' is mostly emotionless and a...   6/10   fresh   
21   3  In the past, Cronenberg has used such deadpan ...    1/4  rotten   
22   3  ... a movie about a sentient zombie, trapped i...      8   fresh   

             critic  top_critic                        publisher  \
0        PJ Nabarro           0                  Patrick Nabarro   
6    Eric D. Snider           0                  EricDSnider.com   
7      Matt Kelemen           0               Las Vegas CityLife   
11     Emanuel Levy           0                  EmanuelLevy.Com   
12   Christian Toto           0                    Big Hollywood   
13     Robert Roten           0              Laramie Movie Scope   
14      Marty Mapes           0                      Movie Habit   
15        Adam Ross           0                   The Aristocrat   
16    Patrick Kolan           0                   Shotgun Cinema   
17  Dennis Schwartz           0        Ozus' World Movie Reviews   
18    Sean P. Means           0                Salt Lake Tribune   
19     John Beifuss           0  Commercial Appeal (Memphis, TN)   
20      Chris Sawin           0                     Examiner.com   
21   Jason Buchanan           0                         TV Guide   
22    Philip Martin           0        Arkansas Democrat-Gazette   

                  date  
0    November 10, 2018  
6        July 17, 2013  
7       April 21, 2013  
11    February 3, 2013  
12    January 15, 2013  
13     January 7, 2013  
14    October 20, 2012  
15  September 27, 2012  
16  September 26, 2012  
17  September 25, 2012  
18  September 14, 2012  
19  September 10, 2012  
20   September 7, 2012  
21   September 7, 2012  
22   September 7, 2012

In [20]:
df_reviews_edit.tail(15)

id                                             review rating   fresh  \
54413  2000  Wasabi does its job colorfully and entertainin...     C+   fresh   
54414  2000  A cheerful enough but imminently forgettable r...  2.5/4  rotten   
54415  2000                             Fluffy and disposible.  2.5/5  rotten   
54416  2000  One minute, you think you're watching a seriou...    2/4  rotten   
54418  2000  Slight, but colorful and forward-moving... Was...     C+   fresh   
54419  2000            Sleek, shallow, but frequently amusing.  2.5/4   fresh   
54420  2000  The spaniel-eyed Jean Reno infuses Hubert with...    3/4   fresh   
54421  2000  Manages to be somewhat well-acted, not badly a...  1.5/4  rotten   
54422  2000  Arguably the best script that Besson has writt...  3.5/5   fresh   
54423  2000          Despite the title, Wasabi isn't so spicy.      C  rotten   
54424  2000  Dawdles and drags when it should pop; it doesn...  1.5/5  rotten   
54428  2000                                                NaN    1/5  rotten   
54429  2000                                                NaN    2/5  rotten   
54430  2000                                                NaN  2.5/5  rotten   
54431  2000                                                NaN    3/5   fresh   

                  critic  top_critic                   publisher  \
54413       Sean Axmaker           0  Seattle Post-Intelligencer   
54414       Mary F. Pols           0          Contra Costa Times   
54415  Maitland McDonagh           0                    TV Guide   
54416        Tor Thorsen           0                    Reel.com   
54418        Brent Simon           0         Entertainment Today   
54419       Gene Seymour           1                     Newsday   
54420       Megan Turner           1               New York Post   
54421        Bob Strauss           0      Los Angeles Daily News   
54422         Wade Major           0          Boxoffice Magazine   
54423                NaN           0                   E! Online   
54424     Manohla Dargis           1           Los Angeles Times   
54428  Michael Szymanski           0                  Zap2it.com   
54429       Emanuel Levy           0             EmanuelLevy.Com   
54430   Christopher Null           0              Filmcritic.com   
54431    Nicolas Lacroix           0                Showbizz.net   

                     date  
54413     October 3, 2002  
54414     October 3, 2002  
54415     October 2, 2002  
54416     October 2, 2002  
54418  September 30, 2002  
54419  September 27, 2002  
54420  September 27, 2002  
54421  September 27, 2002  
54422  September 27, 2002  
54423  September 27, 2002  
54424  September 26, 2002  
54428  September 21, 2005  
54429       July 17, 2005  
54430   September 7, 2003  
54431   November 12, 2002

# Summary of Rotten Tomatos 

## rt.movie_info_info.tsv

This data set has alot of NaN. Initially you start off with having 1560 rows but as you can see from the orginial df there is only 340 results of the the series "box office". 
I was thinking we could use the 340 results and join them with the other df from the other data sets. 
However, 'box_office' has a data type of objects and I think it would be best if we convert all of that in int. I am not sure if this is right but I pretty sure we will have problems creating visuals if we choose to leave it as objects

## rt.movie_reviews.tsv

This data set is pretty confusing, have to think of a way to reformat rating.

## Steps to take 9/26/20

 * Look into the other data set that is also part of the rt 
 * Convert the data types: change objects into int
 * Look into the last data set to see if I can clean that and see if we can use something that is Useful 

# tn.movie_budgets.csv

In [13]:
df_movie_budgets = pd.read_csv('tn.movie_budgets.csv', sep=',', error_bad_lines=False)
df_movie_budgets.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [14]:
df_movie_budgets.columns

Index(['id', 'release_date', 'movie', 'production_budget', 'domestic_gross',
       'worldwide_gross'],
      dtype='object')

In [15]:
df_movie_budgets.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
id                   5782 non-null int64
release_date         5782 non-null object
movie                5782 non-null object
production_budget    5782 non-null object
domestic_gross       5782 non-null object
worldwide_gross      5782 non-null object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


# tmdb.movies.csv

In [16]:
df_tmdb_movie = pd.read_csv('tmdb.movies.csv', sep=',', error_bad_lines=False)
df_tmdb_movie.head()

Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [1]:
from requests import get 
url = 'https://www.imdb.com/search/title/?title_type=tv_series'
url1= 'https://www.imdb.com/search/title/?title_type=tv_series&ref_=adv_prv'
url2= 'https://www.imdb.com/search/title/?title_type=tv_series&start=51&ref_=adv_nxt'
response = get(url)
response1 = get(url1)
print(response1.text[:500])




<!DOCTYPE html>
<html
    xmlns:og="http://ogp.me/ns#"
    xmlns:fb="http://www.facebook.com/2008/fbml">
    <head>
         
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">

    <meta name="apple-itunes-app" content="app-id=342792525, app-argument=imdb:///?src=mdot">



        <script type="text/javascript">var IMDbTimer={starttime: new Date().getTime(),pt:'java'};</script>

<script>
    if (typeof uet == 'function') {
      uet("bb", "LoadTitle"


In [2]:
from bs4 import BeautifulSoup
html_soup = BeautifulSoup(response1.text, 'html.parser')
type(html_soup)

bs4.BeautifulSoup

In [3]:
tv_series_containers = html_soup.find_all('div', class_= 'lister-item mode-advanced')
print(type(tv_series_containers))
print(len(tv_series_containers))

<class 'bs4.element.ResultSet'>
50


In [4]:
first_container = tv_series_containers[0]

In [5]:
first_name = first_container.h3.a.text
first_name

'Ratched'

In [6]:
first_genre = first_container.p.find('span' , class_ = 'genre')
first_genre = first_genre.text
first_genre

'\nCrime, Drama, Mystery            '

In [7]:
first_year = first_container.h3.find('span' , class_='lister-item-year text-muted unbold')
first_year = first_year.text
first_year

'(2020– )'

In [8]:
first_imdb = float(first_container.strong.text)
first_imdb

7.4

In [9]:
first_votes = first_container.find('span' , attrs = {'name':'nv'})
first_votes = int(first_votes['data-value'])
first_votes

16881

In [10]:
names = []
genres = []
years = []
imdb_ratings = []
votes = []

for container in tv_series_containers:
    if container.strong is not None:
    
        name = container.h3.a.text
        names.append(name)
    
        genre = container.p.find('span' , class_ = 'genre').text
        genres.append(genre)
    
        year = container.h3.find('span' , class_='lister-item-year text-muted unbold').text
        years.append(year)
    
        imdb = float(container.strong.text)
        imdb_ratings.append(imdb)
    
        vote = container.find('span' , attrs = {'name':'nv'})['data-value']
        votes.append(int(vote))
    

In [17]:
pages = [str(i) for i in range(51,192263,50)]

In [21]:
from time import sleep
from random import randint

In [29]:
from time import time
start_time = time()

In [30]:
requests = 0
for _ in range(5):
# A request would go here
    requests += 1
    sleep(randint(1,3))
    elapsed_time = time() - start_time
    print('Request: {}; Frequency: {} requests/s'.format(requests, requests/elapsed_time))

Request: 1; Frequency: 0.2159599741999569 requests/s
Request: 2; Frequency: 0.26200426166456786 requests/s
Request: 3; Frequency: 0.28209922931646453 requests/s
Request: 4; Frequency: 0.29336332640527785 requests/s
Request: 5; Frequency: 0.3005456966817158 requests/s


In [31]:
from warnings import warn
warn('Warning Simulation')

C:\Users\egust\anaconda3\envs\learn-env\lib\site-packages\ipykernel_launcher.py:2: UserWarning: Warning Simulation
  


In [33]:
headers = {"Accept-Language": "en-US, en;q=0.5"}

In [36]:
from IPython.core.display import clear_output
start_time = time()
requests = 0
for _ in range(5):
# A request would go here
    requests += 1
    sleep(randint(1,3))
    current_time = time()
    elapsed_time = current_time - start_time
    print('Request: {}; Frequency: {} requests/s'.format(requests, requests/elapsed_time))
clear_output(wait = True)

Request: 1; Frequency: 0.33328507380449995 requests/s
Request: 2; Frequency: 0.39990650455513793 requests/s
Request: 3; Frequency: 0.4283417065310125 requests/s
Request: 4; Frequency: 0.39976063755389385 requests/s
Request: 5; Frequency: 0.4164424173268316 requests/s


In [92]:
names = []
genres = []
years = []
imdb_ratings = []
votes = []
pages = [str(i) for i in range(51,192379,50)]

start_time = time()
requests = 0

for page in pages:
    
    response = get('https://www.imdb.com/search/title/?title_type=tv_series&start='+ page +'&ref_=adv_nxt', headers = headers)
    
    sleep(randint(8,15))

    requests += 1
    elapsed_time = time() - start_time
    print('Request:{}; Frequency: {} requests/s'.format(requests, requests/elapsed_time))
    clear_output(wait = True)

    if response.status_code != 200:
            warn('Request: {}; Status code: {}'.format(requests, response.status_code))

    if requests > 1000:
            warn('Number of requests was greater than expected.')
            break

    html_soup = BeautifulSoup(response.text, 'html.parser')

    tv_series_containers = html_soup.find_all('div', class_= 'lister-item mode-advanced')

    for container in tv_series_containers:
        if container.strong is not None:
            if container.p.find('span' , class_ = 'genre') is not None:
    
                name = container.h3.a.text
                names.append(name)
    
                year = container.h3.find('span' , class_='lister-item-year text-muted unbold').text
                years.append(year)
    
                imdb = float(container.strong.text)
                imdb_ratings.append(imdb)
    
                vote = container.find('span' , attrs = {'name':'nv'})['data-value']
                votes.append(int(vote))
                 
                genre = container.p.find('span' , class_ = 'genre').text
                genres.append(genre)

C:\Users\egust\anaconda3\envs\learn-env\lib\site-packages\ipykernel_launcher.py:26: UserWarning: Number of requests was greater than expected.


In [93]:
len(names)

47927

In [84]:
genres

['\nCrime, Drama, Mystery            ',
 '\nComedy, Family, Fantasy            ',
 '\nCrime, Drama, Thriller            ',
 '\nDrama, Sci-Fi, Thriller            ',
 '\nCrime, Drama, Thriller            ',
 '\nCrime, Drama            ',
 '\nCrime, Drama, Mystery            ',
 '\nComedy, Crime            ',
 '\nAction, Crime, Drama            ',
 '\nDrama, Sci-Fi, Thriller            ',
 '\nCrime, Drama, Thriller            ',
 '\nComedy, Drama            ',
 '\nDrama, Fantasy, Horror            ',
 '\nComedy, Drama, Fantasy            ',
 '\nComedy, Drama            ',
 '\nAction, Crime, Mystery            ',
 '\nCrime, Drama            ',
 '\nAdventure, Drama, Family            ',
 '\nComedy, Drama, Romance            ',
 '\nAction, Adventure, Drama            ',
 '\nDrama, Sci-Fi            ',
 '\nDrama            ',
 '\nComedy            ',
 '\nDrama, Mystery, Sci-Fi            ',
 '\nAction, Drama, Fantasy            ',
 '\nAction, Drama, Fantasy            ',
 '\nAction, Crime, D

In [94]:

genres_list = []

for line in genres:
    genres_list.append(line.strip().split('\n'))

In [95]:
genres_list1 = []

for line in genres_list:
    genres_list1.append(line[0])

In [96]:
len(genres_list1)

47927

In [ ]:
## AttributeError                            Traceback (most recent call last)
#<ipython-input-34-8bc1b5e71eda> in <module>
  #   15     years.append(year)
   #  16 
#---> 17     imdb = float(container.strong.text)
 #    18     imdb_ratings.append(imdb)
  #   19 

#AttributeError: 'NoneType' object has no attribute 'text'

# This means that when iterating there was missing data for the imdb


In [97]:
import pandas as pd 
test_tv_series_df = pd.DataFrame({'tv_show': names,
                                  'genre': genres_list1,
                                  'year': years,
                                  'imdb': imdb_ratings,
                                  'votes': votes})


In [98]:
test_tv_series_df.head(10)

tv_show                    genre         year  \
0                           Sherlock    Crime, Drama, Mystery  (2010–2017)   
1             Julie and the Phantoms  Comedy, Family, Fantasy     (2020– )   
2                              Ozark   Crime, Drama, Thriller     (2017– )   
3                       Black Mirror  Drama, Sci-Fi, Thriller     (2011– )   
4                    Sons of Anarchy   Crime, Drama, Thriller  (2008–2014)   
5                       The Sopranos             Crime, Drama  (1999–2007)   
6  Law & Order: Special Victims Unit    Crime, Drama, Mystery     (1999– )   
7                 Brooklyn Nine-Nine            Comedy, Crime     (2013– )   
8                               NCIS     Action, Crime, Drama     (2003– )   
9                The Handmaid's Tale  Drama, Sci-Fi, Thriller     (2017– )   

   imdb   votes  
0   9.1  781937  
1   8.3    2883  
2   8.4  172935  
3   8.8  421268  
4   8.6  251613  
5   9.2  289081  
6   8.0   77867  
7   8.4  208706  
8   7.8  115966  
9   8.5  161806

In [99]:
test_tv_series_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 47927 entries, 0 to 47926
Data columns (total 5 columns):
tv_show    47927 non-null object
genre      47927 non-null object
year       47927 non-null object
imdb       47927 non-null float64
votes      47927 non-null int64
dtypes: float64(1), int64(1), object(3)
memory usage: 1.8+ MB


In [100]:
test_tv_series_df.to_csv('imbd_tv_series.csv')